src/westpa/core/we_driver.py

```diff
@@ -559,19 +594,33 @@ def _adjust_count(self, bin, groups, target_count):
                 bin.remove(segments[-1])
                 i.remove(segments[-1])
                 new_segments_list = self._split_walker(segments[-1], 2, bin)
+                # for j in new_segments_list:
+                #   print(j.weight)
+                # print(self.smallest_allowed_weight)
+                if all(new_segment.weight < self.smallest_allowed_weight for new_segment in new_segments_list):
+                    #    print("instance of threshold break (ac)")
+                    bin.add(segments[-1])
+                    i.add(segments[-1])
+                else:
+                    #    print("no threshold break (ac)")
+                    i.update(new_segments_list)
+                    bin.update(new_segments_list)
+
                 # KFW CHECK BACK                for new_segment in new_segments_list:
                 # KFW CHECK BACK                    try:
                 # KFW CHECK BACK                        new_segment.id_hist = list(segments[-1].id_hist)
                 # KFW CHECK BACK                    except AttributeError:
                 # KFW CHECK BACK                        pass
-                i.update(new_segments_list)
-                bin.update(new_segments_list)
 
                 if len(bin) == target_count:
                     break
+                elif i == sorted_groups[-1]:
```

Comment:
The unit test is failing because of this line. The original code will run a few iterations of the `while` loop (and therefore, several times of the for-loop inside) to adjust the count to the target value, whereas this line will set the target count to `len(bin)` and break too early.